src/lib/map.js

```diff
@@ -83,28 +93,64 @@ export const createMap = ({containerId, coords, onVehicleClick, onMoveEnd}) => {
         'icon-image': 'drone',
       }
     });
+    map.addSource('pickup', {
+      'type': 'geojson',
+      'data': {
+        'type': 'FeatureCollection',
+        'features': []
+      }
+    });
+    map.addLayer({
+      'id': 'pickup',
+      'type': 'symbol',
+      'source': 'pickup',
+      'minzoom': 10,
+      'layout': {
+        'icon-image': 'pickup',
+      }
+    });
+    map.addSource('dropoff', {
+      'type': 'geojson',
+      'data': {
+        'type': 'FeatureCollection',
+        'features': []
+      }
+    });
+    map.addLayer({
+      'id': 'dropoff',
+      'type': 'symbol',
+      'source': 'dropoff',
+      'minzoom': 10,
+      'layout': {
+        'icon-image': 'dropoff',
+      }
+    });
     map.on('click', 'vehicles', (e) => onVehicleClick(e.features[0].properties.id));
   });
 
   map.on('moveend', () => {
     const mapCenter = map.getCenter();
-    onMoveEnd({ lat: mapCenter.lat, long: mapCenter.lng });
+    onMoveEnd({lat: mapCenter.lat, long: mapCenter.lng});
   });
 
   // Check if user has already granted permission to access geolocation
   // If permission was granted, get user location and center map on them
   hasGeolocationPermission()
     .then(getUserLocation)
     .then(
-      ({ coords }) => map.setCenter([coords.longitude, coords.latitude])
+      ({coords}) => map.setCenter([coords.longitude, coords.latitude])
     ).catch(() => {});
 
   return map;
 };
 
-export const updateMap = (map, vehicles = []) => {
+export const updateMap = (map, vehicles = [], {pickup, dropoff} = {}) => {
   handleMapUpdate(map, () => {
-    map.getSource('vehicles').setData(createGeoJson(vehicles));
+    if (vehicles) map.getSource('vehicles').setData(createGeoJson(vehicles));
+    if (pickup && dropoff) {
+      map.getSource('pickup').setData(turf.point([pickup.long, pickup.lat]));
```

Comment:
I sometimes get a fatal error on this line.
Usually if I start searching and then click the back button while it is searching.

I am assuming this has something to do with `clearPins()` being called and then this line trying to get the `source`.

<img width="685" alt="clearpins" src="https://user-images.githubusercontent.com/926653/33528912-932d9f58-d871-11e7-8b81-4b5154e85656.png">

Reply:
Yeah, you're right. That's the issue.